aspnetcore/security/authentication/claims.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Mapping claims, customizing claims and transforming claims in ASP.NET Core
+author: damienbod
+description: Learn how to map claims, do claims transformations, customize claims.
+monikerRange: '>= aspnetcore-3.1'
+ms.author: riande
+ms.custom: mvc
+ms.date: 05/29/2021
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: security/authentication/claims
+---
+# Mapping claims, customizing claims and transforming claims in ASP.NET Core
+
+By [Damien Bowden](https://github.com/damienbod)
+
+Claims can be created from any user data or identity data which can be issued using a trusted identity provider or ASP.NET Core identity. A claim is a name value pair that represents what the subject is, not what the subject can do.
+This article covers the following areas:
+
+* How to configure and map claims using an OpenID Connect client
+* Set the name claim and the role claim
+* Reset the claims namespaces
+* Customize, extend the claims using IClaimsTransformation
+
+## Mapping claims using Open ID Connect authentication
+
+The profile claims can be returned in the **id_token** which is returned after a successful authentication. The ASP.NET Core client application just needs to request the profile scope. When using the id_token for claims, no extra claims mapping is required.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddAuthentication(options =>
+    {
+        options.DefaultScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+        options.DefaultChallengeScheme = OpenIdConnectDefaults.AuthenticationScheme;
+    })
+   .AddCookie()
+   .AddOpenIdConnect(options =>
+   {
+       options.SignInScheme = "Cookies";
+       options.Authority = "-your-identity-provider-";
+       options.RequireHttpsMetadata = true;
+       options.ClientId = "-your-clientid-";
+       options.ClientSecret = "-your-client-secret-from-user-secrets-or-keyvault";
+       options.ResponseType = "code";
+       options.UsePkce = true;
+       options.Scope.Add("profile");
+       options.SaveTokens = true;
+   });
+```
+
+A second way to get the user claims is to use the OpenID Connect User Info API. The ASP.NET Core client application uses the **GetClaimsFromUserInfoEndpoint** property to configure this. One important difference from the first settings, is that you must specify the claims you require using the **MapUniqueJsonKey** method, otherwise only the **name**, **given_name** and **email** standard claims will be available in the client application. The claims included in the id_token are mapped per default. This is the major difference to the first option. You must explicitly define some of the claims you require.
```

Comment:
```suggestion
Another way to get the user claims is to use the OpenID Connect User Info API. The ASP.NET Core client application uses the `GetClaimsFromUserInfoEndpoint` property to configure this. One important difference from the first settings, is that you must specify the claims you require using the `MapUniqueJsonKey` method, otherwise only the `name`, `given_name` and `email` standard claims will be available in the client application. The claims included in the `id_token` are mapped per default. This is the major difference to the first option. You must explicitly define some of the claims you require.
```

Reply:
fixed